src/components/number-picker/number-picker.component.spec.ts

```diff
@@ -430,4 +436,45 @@ describe('Number Picker Component - value', () => {
         expect(numberPicker.classList.contains('ng-invalid')).toBe(false);
     });
 
+    it('should call the event emitter once per change', async () => {
+        const controlUp = nativeElement.querySelector<HTMLElement>('.number-picker-control-up');
+
+        spyOn(component, 'onValueChange');
+
+        controlUp.click();
+
+        expect(component.onValueChange).toHaveBeenCalledWith(1);
+        expect(component.onValueChange).toHaveBeenCalledTimes(1);
+
+    });
+
+    it('should not emit valueChange when the value input changes', async () => {
+        component.value = 1;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        spyOn(component, 'onValueChange');
+
+        let controlUp = nativeElement.querySelector<HTMLButtonElement>('input').value;
+        expect(controlUp).toEqual('1');
+
+        expect(component.onValueChange).not.toHaveBeenCalled();
+    });
+
+    it('should emit valueChange when entering value' , () => {
+        fixture.detectChanges();
+
+        const inputChange = getInput();
+        inputChange.focus();
+
+        dispatchKeyboardEvent(inputChange, 'keydown', 0, null, '0');
+        component.value = 0;
+        fixture.detectChanges();
```

Comment:
As we discussed, this test has no assertions so we should add some in. Additionally we should also test that ngModelChange is emitted here as well as valueChange.